include/libtorrent/aux_/session_impl.hpp

```diff
@@ -619,10 +619,11 @@ namespace libtorrent
 				, sha1_hash const& sent_target, udp::endpoint const& ep) override;
 
 #ifndef TORRENT_DISABLE_LOGGING
-			virtual void log(libtorrent::dht::dht_logger::module_t m, char const* fmt, ...)
-				override TORRENT_FORMAT(3,4);
+			virtual bool should_log(module_t m) const override;
+			virtual void log(module_t m, char const* fmt, ...)
+				const override TORRENT_FORMAT(3,4);
 			virtual void log_packet(message_direction_t dir, char const* pkt, int len
-				, udp::endpoint node) override;
+				, udp::endpoint node) const override;
```

Comment:
it's not obvious to me what this "const" buy you. for instance, it would make it less convenient to create an observer that store loglines an a std::vector<std::string> member variable (since it would have to be mutable or const_cast<>ed)